quarto/_extensions/countdown/countdown.lua

```diff
@@ -0,0 +1,353 @@
+  end
+
+  return { minutes = tonumber(minutes), seconds = tonumber(seconds) }
+end
+
+local function countdown(args, kwargs, meta)
+  local minutes, seconds
+
+  -- Retrieve named time arguments and fallback on default values if missing
+  arg_time = parseTimeString(args)
+  if isVariablePopulated(arg_time) then
+    minutes = arg_time.minutes
+    seconds = arg_time.seconds
+    if isVariablePopulated(tryOption(kwargs, "minutes")) or 
+       isVariablePopulated(tryOption(kwargs, "seconds")) then
+      quarto.log.warning(
+        "Please do not specify `minutes` or `seconds` parameters" ..
+        "when using the time string format.")
+    end
+  else
+    minutes = tonumber(getOption(kwargs, "minutes", 1))
+    seconds = tonumber(getOption(kwargs, "seconds", 0))
+  end
+
+  -- Calculate total time in seconds
+  local time = minutes * 60 + seconds
+
+  -- Calculate minutes by dividing total time by 60 and rounding down
+  minutes = math.floor(time / 60)
+
+  -- Calculate remaining seconds after extracting minutes
+  seconds = time - minutes * 60
+  
+  -- Check if minutes is greater than or equal to 100 (the maximum possible for display)
+  if minutes >= 100 then
+    quarto.log.error("The number of minutes must be less than 100.")
+    assert("true" == "false")
+  end
+
+  if needsToExportDependencies then
+    ensureHTMLDependency(meta)
+  end
+
+  -- Retrieve the ID given by the user or attempt to create a unique ID by timestamp
+  local id = getOption(kwargs, "id", "timer_" .. pandoc.utils.sha1(tostring(os.time())))
+
+  -- Construct the 'class' attribute by appending "countdown" to the existing class (if any)
+  local class = getOption(kwargs, "class", "")
+  class = class ~= "" and "countdown " .. class or "countdown"
+
+  -- Determine if a warning should be given
+  local warn_when = tonumber(getOption(kwargs, "warn_when", 0))
+    
+  -- Retrieve and convert "update_every" attribute to a number, default to 1 if not present or invalid
+  local update_every = tonumber(getOption(kwargs, "update_every", 1))
+
+  -- Retrieve "blink_colon" attribute and set 'blink_colon' to true if it equals "true", otherwise false
+  local blink_colon = getOption(kwargs, "blink_colon", update_every > 1) == "true"
+
+  -- Retrieve "start_immediately" attribute and set 'start_immediately' to true if it equals "true", otherwise false
+  local start_immediately = getOption(kwargs, "start_immediately", "false") == "true"
+
+  -- Retrieve "play_sound" attribute as a string, default to "false" if not present
+  local play_sound = tryPlaySound(getOption(kwargs, "play_sound", "false"))
+
+  -- Check to see if positional outcomes are set; if not, default both bottom and right to 0.
+  if isVariableEmpty(tryOption(kwargs, "top"))   and
+     isVariableEmpty(tryOption(kwargs, "left"))  and
+     isVariableEmpty(tryOption(kwargs, "right")) and
+     isVariableEmpty(tryOption(kwargs, "bottom")) then
+    kwargs["bottom"] = 0 
+    kwargs["right"] = 0 
+  end
```

Comment:
```suggestion
  if isVariableEmpty(tryOption(kwargs, "top")) and
     isVariableEmpty(tryOption(kwargs, "bottom")) then
    kwargs["bottom"] = 0
  end    
     
  if isVariableEmpty(tryOption(kwargs, "left"))  and
     isVariableEmpty(tryOption(kwargs, "right")) then
    kwargs["right"] = 0 
  end
```

This better follows the R logic which is hidden in the function signature: https://github.com/gadenbuie/countdown/blob/34204099a9b575b72504523f739a51c200bbca5f/r/R/countdown.R#L151-L152

This makes it a lot easier to move the timer around:

```
<!-- bottom right -->
{{< countdown 1:30 >}}

<!-- bottom left -->
{{< countdown 1:30 left=0 >}}

<!-- top right -->
{{< countdown 1:30 top=0 >}}

<!-- top left -->
{{< countdown 1:30 top=0 left=0 >}}
```